src/server/game/Entities/Player/Player.cpp

```diff
@@ -5110,6 +5116,20 @@ void Player::UpdateRating(CombatRating cr)
     }
 }
 
+void Player::UpdateRatingBonus(CombatRating cr, bool apply)
+{
+    int32 bonus = 0;
+    int32 amount = m_baseRatingValue[cr];
+    // Apply bonus from SPELL_AURA_MOD_RATING_PCT
+    // stat used stored in miscValueA for this aura
+    AuraEffectList const& modRatingPct = GetAuraEffectsByType(SPELL_AURA_MOD_RATING_PCT);
+    for (AuraEffectList::const_iterator i = modRatingPct.begin(); i != modRatingPct.end(); ++i)
+        if ((*i)->GetMiscValue() & (1 << cr))
+            bonus += int32(CalculatePct(amount, (*i)->GetAmount()));
+
+    m_bonusRatingValue[cr] = (apply ? bonus : 0);
```

Comment:
Pretty sure this doesn't work as you want it for !apply case when the player had more than 1 aura with SPELL_AURA_MOD_RATING_PCT